src/Tickets/Commerce/Hooks.php

```diff
@@ -143,6 +145,54 @@ protected function add_filters() {
 		add_action( 'tribe_editor_config', [ $this, 'filter_tickets_editor_config' ] );
 
 		add_filter( 'wp_list_table_class_name', [ $this, 'filter_wp_list_table_class_name' ], 10, 2 );
+
+		add_filter( 'tribe_dropdown_tec_tc_order_table_events', [ $this, 'provide_events_results_to_ajax' ], 10, 2 );
+	}
+
+	/**
+	 * Provides the results for the events dropdown in the Orders table.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $results The results.
+	 * @param array $search The search.
+	 *
+	 * @return array
+	 */
+	public function provide_events_results_to_ajax( $results, $search ) {
+		if ( empty( $search['term'] ) ) {
+			return $results;
+		}
+
+		$term = $search['term'];
+
+		$args = [
+			'no_found_rows'          => true,
+			'update_post_meta_cache' => false,
+			'update_post_term_cache' => false,
+			'post_type'              => (array) tribe_get_option( 'ticket-enabled-post-types', [] ),
+			'post_status'            => 'any',
+			'posts_per_page'         => 10,
+			's'                      => $term,
+		];
+
+		$query = new WP_Query( $args );
+
+		if ( empty( $query->posts ) ) {
+			return $results;
+		}
+
+		$results = array_map(
+			function ( $r ) {
```

Comment:
This is a throwaway anonymous function, but it would still be better to use a variable name that makes it clear what the data is. Even more ideal would be to type-hint `WP_Post` as the parameter type.